sys/log/full/src/log.c

```diff
@@ -1120,6 +1337,35 @@ log_set_max_entry_len(struct log *log, uint16_t max_entry_len)
     log->l_max_entry_len = max_entry_len;
 }
 
+int
+log_fill_num_entries(struct log *log, const void *dptr,
+                     struct log_entry_hdr *hdr, uint16_t offset)
+{
+#if MYNEWT_VAL(LOG_FLAGS_TLV_SUPPORT) && MYNEWT_VAL(LOG_TLV_NUM_ENTRIES)
+    int rc = 0;
+    struct log_tlv tlv = {0};
+
+    rc = log_read(log, dptr, &tlv, offset, sizeof(struct log_tlv));
+    if (rc < sizeof(struct log_tlv)) {
+        return rc;
+    }
+
+    offset += sizeof(struct log_tlv);
+
+    rc = log_read(log, dptr, &hdr->ue_num_entries, offset,
+                  LOG_NUM_ENTRIES_SIZE);
+    if (rc >= LOG_NUM_ENTRIES_SIZE) {
+        return SYS_EOK;
+    } else {
+        return rc;
+    }
+#else
+    hdr->ue_num_entries = 0;
+
+    return SYS_ENOTSUP;
+#endif
+}
```

Comment:
Function assumes that LTV for num entries is at fixed position from the end of fcb entry. Why use TLV if it's not even checked. Function could skip reading 2 bytes of TLV to save some time and code.